src/aggregatederror.ts

```diff
@@ -1,22 +1,9 @@
+import * as types from "./types";
+
 /** @hidden */
 const separator = "-----------------------------------";
 
-/**
- * A specific error thrown when context.df.Task.all() fails. Its message
- * contains an aggregation of all the exceptions that failed. It should follow the
- * below format:
- *
- * context.df.Task.all() encountered the below error messages:
- *
- * Name: DurableError
- * Message: The activity function "ActivityA" failed.
- * StackTrace: <stacktrace>
- * -----------------------------------
- * Name: DurableError
- * Message: The activity function "ActivityB" failed.
- * StackTrace: <stacktrace>
- */
-export class AggregatedError extends Error {
+export class AggregatedError extends Error implements types.AggregatedError {

```

Comment:
I notice this pattern of having a class "X" implement "types.X". Is this a common pattern in Node? I find it a bit confusing to have two components of the same name

Reply:
Yup, this is a common pattern. This is the distinction between class declaration and implementation, since we want to move towards manual authoring of `d.ts` files. This is taking us one step in this direction. In reality, these components already exist, except the declaration piece is currently automatically generated.

Reply:
Also, you can see this pattern too in `@azure/functions` with classes like `HttpRequest` and `HttpResponse` that it exports.